models/demos/bert_tiny/tests/test_performance.py

```diff
@@ -0,0 +1,136 @@
+    torch_token_type_ids = torch.zeros((batch_size, sequence_size), dtype=torch.int32)
+    torch_position_ids = torch.zeros((batch_size, sequence_size), dtype=torch.int32)
+    torch_attention_mask = torch.zeros(1, sequence_size)
+
+    parameters = preprocess_model_parameters(
+        initialize_model=lambda: pytorch_model,
+        device=device,
+        convert_to_ttnn=lambda *_: True,
+    )
+
+    ttnn_bert_inputs = ttnn.from_torch(torch_bert_input, dtype=ttnn.uint32, device=device)
+    ttnn_token_type_ids = ttnn.from_torch(torch_token_type_ids, dtype=ttnn.uint32, device=device)
+    ttnn_position_ids = ttnn.from_torch(torch_position_ids, dtype=ttnn.uint32, device=device)
+    ttnn_attention_mask = ttnn.from_torch(torch_attention_mask, dtype=ttnn.bfloat16, device=device)
+
+    durations = []
+    for i in range(2):
+        start = time.time()
+        ttnn_output = bert_for_question_answering(
+            config,
+            input_ids=ttnn_bert_inputs,
+            token_type_ids=ttnn_token_type_ids,
+            position_ids=ttnn_position_ids,
+            attention_mask=ttnn_attention_mask,
+            parameters=parameters,
+            device=device,
+        )
+        output = ttnn.from_device(ttnn_output)
+
+        end = time.time()
+        durations.append(end - start)
+        enable_persistent_kernel_cache()
+
+    inference_and_compile_time, inference_time, *_ = durations
+
+    expected_compile_time, expected_inference_time = get_expected_times("bert_tiny")
+    prep_perf_report(
+        model_name="bert_tiny",
+        batch_size=batch_size,
+        inference_and_compile_time=inference_and_compile_time,
+        inference_time=inference_time,
+        expected_compile_time=expected_compile_time,
+        expected_inference_time=expected_inference_time,
+        comments="",
+        inference_time_cpu=0.0,
+    )
+
+    logger.info(f"Compile time: {inference_and_compile_time - inference_time}")
+    logger.info(f"Inference time: {inference_time}")
+    logger.info(f"Samples per second: {1 / inference_time * batch_size}")
+
+
+@pytest.mark.models_device_performance_bare_metal
+@pytest.mark.parametrize(
+    "batch_size, expected_perf",
+    [
+        (8, 2680),
+    ],
+)
+def test_perf_device_bare_metal(batch_size, expected_perf):
+    subdir = "ttnn_bert_tiny"
+    num_iterations = 1
+    margin = 0.03
+
+    if is_wormhole_b0():
+        expected_perf = 2680
+    else:
+        expected_perf = 978.62
+
+    command = f"pytest tests/ttnn/integration_tests/bert_tiny/test_bert_tiny.py::test_bert_for_question_answering"
+    cols = ["DEVICE FW", "DEVICE KERNEL", "DEVICE BRISC KERNEL"]
+
+    inference_time_key = "AVG DEVICE KERNEL SAMPLES/S"
+    expected_perf_cols = {inference_time_key: expected_perf}
+
+    post_processed_results = run_device_perf(command, subdir, num_iterations, cols, batch_size)
+    expected_results = check_device_perf(post_processed_results, margin, expected_perf_cols)
```

Comment:
test should fail if perf is slower than expected

Reply:
Device perf will fail by default if perf is slower than expected.

Reply:
No, check_device_perf has assert_on_fail=False by default, you need to set it to True.

Reply:
Updated the code with assert_on_fail=True